crates/q_cli/src/cli/chat/tools/fs_write.rs

```
@@ -121,6 +123,44 @@ impl FsWrite {
                 fs.write(&path, &file).await?;
                 Ok(Default::default())
             },
+            FsWrite::Append { path, new_str } => {
+                let path = sanitize_path_tool_arg(ctx, path);
+
+                // Create the file if it doesn't exist
+                if !fs.exists(&path) {
+                    if let Some(parent) = path.parent() {
+                        fs.create_dir_all(parent).await?;
+                    }
+                }
+
+                queue!(
+                    updates,
+                    style::Print("Appending to: "),
+                    style::SetForegroundColor(Color::Green),
+                    style::Print(format_path(cwd, &path)),
+                    style::ResetColor,
+                    style::Print("\n"),
+                )?;
+
+                // Read existing content if file exists
+                let mut file_content = if fs.exists(&path) {
+                    fs.read_to_string(&path).await.unwrap_or_default()
```

Can we return the error here if we fail reading the file for some reason? Rather append stick to what it's intended to do instead of potentially replacing all of the contents.

---

Sure.  
Tool definition will need to change too for that, as I let the tool know it would create if it doesn't exist.

---

Done.